### PR TITLE
feat(typeorm-legacy): custom user fields

### PR DESCRIPTION
**Pull request by g10**
fix issue https://github.com/nextauthjs/adapters/issues/82

**note:** also migrated the [tutorial](https://github.com/nextauthjs/next-auth/blob/main/www/docs/tutorials/typeorm-custom-models.md) to this repo

### CODE REVIEW

File: packages/typeorm-legacy/src/models/user.js [hunk 1]
```diff
@@ -1,18 +1,9 @@
...
-    }
-    if (emailVerified) {
-      const currentDate = new Date()
-      this.emailVerified = currentDate
-    }
+  constructor(profile) {
+    this.name = profile?.name || undefined
+    this.email = profile?.email || undefined
+    this.image = profile?.image || undefined
+    this.emailVerified = profile?.emailVerified ? new Date() : undefined
```
Comment (balazsorban44) [inline packages/typeorm-legacy/src/models/user.js:6]:
```suggestion
  constructor(profile) {
    this.name = profile?.name
    this.email = profile?.email
    this.image = profile?.image
    this.emailVerified = profile?.emailVerified && new Date()
```

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Thank you! I left a small suggestions. Also for this to work, please fix the merge conflicts by merging the `canary` branch into your own. This adapter has recently been simplified

**Comment by limal** [comment]
This is a must needed addition to custom models in NextAuth.js.

The current `phoneNumber` [tutorial](https://next-auth.js.org/tutorials/typeorm-custom-models) is not helpful as `phoneNumber` is not passed with a profile.

**Review by balazsorban44 — APPROVED** [review]

**Comment by balazsorban44** [comment]
Thanks for the PR! Seems that there is some problem with publishing. Once we figure that out, I think it is good to go!

**Pull request merged**